src/tm_devices/drivers/pi/scopes/tekscope_2k/tekscope_2k.py

```diff
@@ -18,10 +24,117 @@ class TekScope2k(Scope, ABC):
     # Properties
     ################################################################################################
 
+    @cached_property
+    def total_channels(self) -> int:
+        """Return the total number of channels (all types)."""
+        return int(self.model[6])
+
     ################################################################################################
     # Public Methods
     ################################################################################################
 
+    def waveforms_displayed(self) -> List[str]:
+        """Returns list of waveforms currently displayed on oscilloscope screen.
+
+        Args:
+            None
+
+        Returns:
+            List of channel strings
+        """
+        source_string = self.query("SELECT?")
+        source_string = source_string.split(":")[2]  # Remove :SELECT: from beginning
```

Comment:
```suggestion
        source_string = source_string.split(":")[-1]  # Remove :SELECT: from beginning
```

I would recommend grabbing the last item in the list, just in case there are no colons to split on

Reply:
Changed as requested.